### DISCUSSION TITLE
Predicting the output with new values inside a complex domain

### DISCUSSION THREAD

**Discussion by ruchitkini** [lululxvi/deepxde#1502] [General]
Hi LuLu,

I have a doubt regarding predicting the values using the trained model. After training the model for a complex 2D geometry one can predict the output as u = model. predict(points), where points are the new data points.
My question is, how can we set these points to be inside the boundary of the 2D domain as I would be interested only in the new points inside the domain, and then use the results to plot a contour plot, etc.?

One more question, when extracting the points from Gmsh as train points, and after training I tried to plot the test points, they are the same as the train points. Why is that so? Even when I increase the test points, they overlap with the previous points and do not test for new points.

**Comment by lululxvi** [comment]
You can predict for any point you like.

**Reply by ruchitkini** [reply]
Thank you for your response. So now I have trained the model with certain geometry points. If I change the geometry points ( for example: increase or decrease the rectangle dimensions or entirely change to some different geometry), keeping pde and boundary conditions the same, Will I get the accurate solution or do I need to retrain the model for a new geometry?
